crates/turborepo-lib/src/task_hash.rs

```diff
@@ -126,58 +118,74 @@ impl PackageInputsHashes {
                 let scm_telemetry = package_task_event.child();
                 // Try hashing with the daemon, if we have a connection. If we don't, or if we
                 // timeout or get an error, fallback to local hashing
-                let hash_object = daemon
-                    .as_mut()
-                    .and_then(|daemon| {
-                        let handle = handle.clone();
-                        // We need an async block here because the timeout must be created with an
-                        // active tokio context. Constructing it directly in
-                        // the rayon thread doesn't provide one and will crash at runtime.
-                        handle
-                            .block_on(async {
-                                tokio::time::timeout(
-                                    Duration::from_millis(100),
-                                    daemon.get_file_hashes(package_path, &task_definition.inputs),
-                                )
-                                .await
-                            })
-                            .map_err(|e| {
-                                tracing::debug!(
-                                    "daemon file hashing timed out for {}",
-                                    package_path
-                                );
-                                e
-                            })
-                            .ok() // If we timed out, we don't need to error,
-                                  // just return None so we can move on to local
-                    })
-                    .and_then(|result| {
-                        match result {
-                            Ok(hashes_resp) => Some(
-                                hashes_resp
-                                    .file_hashes
-                                    .into_iter()
-                                    .map(|(path, hash)| {
-                                        (
-                                            RelativeUnixPathBuf::new(path)
-                                                .expect("daemon returns relative unix paths"),
-                                            hash,
-                                        )
-                                    })
-                                    .collect::<HashMap<_, _>>(),
-                            ),
-                            Err(e) => {
-                                // Daemon could've failed for various reasons. We can still try
-                                // local hashing.
-                                tracing::debug!(
-                                    "daemon file hashing failed for {}: {}",
-                                    package_path,
+                #[cfg(feature = "daemon-file-hashing")]
+                let hash_object = {
```

Comment:
Super minor, but when possible I think `if cfg!("daemon-file-hashing") {...} else {...}` is better than conditional compilation in situations like this. Too often have we had code that doesn't compile when a feature isn't enabled.

Reply:
do both syntaxes compile away the branch that isn't used? +1 for writing in a way that protects us from code that doesn't compile

Reply:
Yeah I'd be really surprised if LLVM can't figure out a const propagation and dead code elimination